### ISSUE TITLE
message passing implementation

### ISSUE THREAD

**Issue by gyzou** [harryhan618/SCNN_Pytorch#18]
Learned a lot from your code. Is your implementation of message passing the sequential or parallel scheme in the paper? SCNN_U and SCNN_D are the same operation, as well as SCNN_R and SCNN_L?

**Comment by harryhan618** [comment]
What I implemented, as described in the paper, is the sequential scheme. And difference between `SCNN_U ` and `SCNN_D` is that, compared to `SCNN_D`, when doing `SCNN_U` you need to flip the input upside down and flip the output again. Similar thing applied to `SCNN_R` and `SCNN_L`.

**Comment by gyzou** [comment]
Can you post the hyperparameters for the results you published.

I trained 300 epochs with batch size 32, on the tusimple data. I couldn't see anything meaningful.

**Comment by harryhan618** [comment]
Maybe you should try lr in the range of 1e-3 to 1e-2. I remember when I trained the model, I used distributed training with several GPUS (with DistributedDataParallel). Thus the learning rate is quite large.
Sorry I can't help you with the parameter tuning. Only do you dig into your experiments so that you can really know what deep learning is. Have good time.
